zk_prover/src/merkle_sum_tree/mst.rs

```diff
@@ -55,9 +84,50 @@ impl<const N_ASSETS: usize, const N_BYTES: usize> MerkleSumTree<N_ASSETS, N_BYTE
             nodes,
             depth,
             entries,
+            is_sorted,
         })
     }
 
+    /// Updates the balances of the entry with the given username and returns the new root of the tree.
+    ///
+    /// # Arguments
+    ///
+    /// * `username`: The username of the entry to update
+    /// * `new_balances`: The new balances of the entry
+    ///
+    /// # Returns
+    ///
+    /// The new root of the tree
+    pub fn update_leaf(
```

Comment:
Using `update_leaf` as a way to avoid rebuilding the entire Merkle tree each time seems efficient. However, I suspect that updating the tree with `update_leaf` might be slower than `build_tree` when more than half of the user balances are being modified, especially since `build_tree` employs a multi-threaded approach. 
Do you have any references or benchmarks regarding this?

Reply:
Points from discussion at the meeting
- parallelized batch insertion can be a good compromise between a single leaf update and a tree rebuild;
- the relation between the user balance update frequency and the proof solvency round frequency would define the ratio of updated leaves but it is not known yet;
- we can benchmark if we modify the test data to contain only unique usernames.